### PR TITLE
Start and stop Azure Spring Cloud service instance

### PR DESCRIPTION
**Pull request by qingc**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

File: src/spring-cloud/azext_spring_cloud/custom.py [hunk 1]
```diff
@@ -209,6 +209,24 @@ def spring_cloud_delete(cmd, client, resource_group, name, no_wait=False):
     return sdk_no_wait(no_wait, client.begin_delete, resource_group_name=resource_group, service_name=name)
 
 
+def spring_cloud_start(cmd, client, resource_group, name, no_wait=False):
+    resource = client.services.get(resource_group, name)
+    state = resource.properties.provisioning_state
+    power_state = resource.properties.power_state
+    if state != "Succeeded" or power_state != "Stopped":
+        raise CLIError("Service is in Provisioning State({}) and Power State({}), starting cannot be performed.".format(state, power_state))
```
[the remark this answers is not in the document]
Comment (qingc) [reply src/spring-cloud/azext_spring_cloud/custom.py]: Updated. Thanks.

File: src/spring-cloud/azext_spring_cloud/custom.py [hunk 2]
```diff
@@ -209,6 +209,24 @@ def spring_cloud_delete(cmd, client, resource_group, name, no_wait=False):
...
+    return sdk_no_wait(no_wait, client.services.begin_start, resource_group_name=resource_group, service_name=name)
+
+
+def spring_cloud_stop(cmd, client, resource_group, name, no_wait=False):
+    resource = client.services.get(resource_group, name)
+    state = resource.properties.provisioning_state
+    power_state = resource.properties.power_state
+    if state != "Succeeded" or power_state != "Running":
+        raise CLIError("Service is in Provisioning State({}) and Power State({}), stopping cannot be performed.".format(state, power_state))
```
[the remark this answers is not in the document]
Comment (qingc) [reply src/spring-cloud/azext_spring_cloud/custom.py]: Updated.

File: src/spring-cloud/HISTORY.md [hunk 3]
```diff
@@ -1,13 +1,19 @@
...
 -----
 * Add support for Diagnostic Operation. Heap dump: 'spring-cloud app deployment generate-heap-dump'. Thread Dump: 'spring-cloud app deployment generate-thread-dump'. JFR: 'spring-cloud app deployment start-jfr'
 
 2.11.1
 -----
 * Add support for public certificate crud, source could be either key vault or local file
-* Application could load public certificate by using argument `--loaded_public_certificate_file` in batch or 
+* Application could load public certificate by using argument `--loaded_public_certificate_file` in batch or
   directly using `spring-cloud app append-loaded-public-certificate` one by one
 * Add support to list all apps which have loaded the certificate `spring-cloud certificate list-reference-app`
```
Comment (zhoxing-ms) [inline src/spring-cloud/HISTORY.md:18]: Here are two descriptions for version `2.11.1`, which are repeated

File: src/spring-cloud/setup.py [hunk 4]
```diff
@@ -16,7 +16,7 @@
 
 # TODO: Confirm this is the right version number you want and it matches your
 # HISTORY.rst entry.
-VERSION = '2.11.1'
+VERSION = '2.11.3'
```
Comment (zhoxing-ms) [inline src/spring-cloud/setup.py:19]: Why is the latest version of `HISTORY.md` `2.11.2`, but the version here is `2.11.3`?

File: src/spring-cloud/HISTORY.md (hunk 3 above)
Comment (qingc) [reply src/spring-cloud/HISTORY.md]: Updated, thanks!

File: src/spring-cloud/setup.py (hunk 4 above)
Comment (zhoxing-ms) [reply src/spring-cloud/setup.py:19]: If the version of `2.11.2` has not been released before, it is recommended not to skip the release of version `2.11.2`

Comment (qingc) [reply src/spring-cloud/setup.py:19]: That's because we missed `2.11.2` in `HISTORY.md` in the last commit.

Comment (qingc) [reply src/spring-cloud/setup.py]: > If the version of `2.11.2` has not been released before, it is recommended not to skip the release of version `2.11.2`. Maybe we can consider merging the history of two `2.11.1` version, and release this PR as version `2.11.2`

Make sense, updated to `2.11.2` now. Please help to check.

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**